pkgs/by-name/th/thc-secure-delete/package.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, stdenvNoCC # x86_64-darwin and aarch64-darwin cant find C compiler
+, fetchFromGitHub
+, pkg-config
+, gcc
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "thc-secure-delete";
+  version = "3.1.1";
+
+  src = fetchFromGitHub {
+    owner = "gordrs";
+    repo = "thc-secure-delete";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-hvWuxFkzhOSCplPtyjRtn36bIk6KdPBcpr3lAmiAyfE=";
+  };
+
+  build-systems = [ pkg-config gcc ];
+
+  buildPhase = ''
```

Comment:
If upstream supports building with clang, it should probably be substituting `$(CC)` for GCC instead of using the path to GCC. Also, substituting a dependency directly into the build phase probably breaks cross-compilation and definitely prevents `overrideSDK` from working should it be needed in the future.

Reply:
This should also probably be overriding `postPatch`, so it can use the `buildPhase` for `make`.

Reply:
And `--replace-fail` instead of `--replace` in case the `Makefile` changes and the substitution stops happening.

Reply:
And if it doesn't support building with clang, do the same as @reckenrode said, but use gccStdenv instead of stdenv. You should not use stdenvNoCC here in any case.

Reply:
It builds with Clang, just needs `CC=${stdenv.cc.targetPrefix}cc` in `makeFlags`.